realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -558,6 +558,41 @@ public void not() {
         }
     }
 
+    @Test
+    public void emptyGroup() {
+        populateTestRealm(); // create TEST_DATA_SIZE objects
+        RealmResults<AllTypes> fives = realm.where(AllTypes.class)
+                .equalTo(AllTypes.FIELD_LONG, 5)
+                .beginGroup()
+                .endGroup()
+                .findAll();
+        // Only on object with value 5
+        assertEquals(1, fives.size());
+    }
+
+    @Test (expected = UnsupportedOperationException.class)
+    public void or_emptyGroupThrows() {
+        populateTestRealm(); // create TEST_DATA_SIZE objects
+        RealmResults<AllTypes> fives = realm.where(AllTypes.class)
```

Comment:
With this pattern, it is difficult to guess exactly which line throws the exception. Normally we try to capture the explicit call in a try/catch. This also means that using the `@Test( ....)` pattern is usually a bad idea.